src/com/amazon/ion/benchmark/WriteRandomIonValues.java

```diff
@@ -218,122 +247,176 @@ private static Integer randomLocalOffset(Random random) {
-                        );
-                        break;
-                    case MINUTE:
-                        timestamp = Timestamp.forMinute(
-                            random.nextInt(9998) + 1,
-                            random.nextInt(12) + 1,
-                            random.nextInt(28) + 1, // Use max 28 for simplicity. Not including up to 31 is not going to affect the measurement.
-                            random.nextInt(24),
-                            random.nextInt(60),
-                            randomLocalOffset(random)
-                        );
-                        break;
-                    case SECOND:
-                        timestamp = Timestamp.forSecond(
-                            random.nextInt(9998) + 1,
-                            random.nextInt(12) + 1,
-                            random.nextInt(28) + 1, // Use max 28 for simplicity. Not including up to 31 is not going to affect the measurement.
-                            random.nextInt(24),
-                            random.nextInt(60),
-                            random.nextInt(60),
-                            randomLocalOffset(random)
-                        );
-                        break;
-                    case FRACTION:
-                        timestamp = Timestamp.forSecond(
-                            random.nextInt(9998) + 1,
-                            random.nextInt(12) + 1,
-                            random.nextInt(28) + 1, // Use max 28 for simplicity. Not including up to 31 is not going to affect the measurement.
-                            random.nextInt(24),
-                            random.nextInt(60),
-                            randomSecondWithFraction(random),
-                            randomLocalOffset(random)
-                        );
-                        break;
-                    default:
-                        throw new IllegalStateException();
-                }
-                writer.writeTimestamp(timestamp);
-            }
-        }
-        System.out.println("Finished writing timestamps. Verifying.");
-        try (IonReader reader = IonReaderBuilder.standard().build(new BufferedInputStream(new FileInputStream(file)))) {
-            int i = 0;
-            while (reader.next() != null) {
-                if (reader.getType() != IonType.TIMESTAMP) {
-                    throw new IllegalStateException("Found non-timestamp");
+//            Timestamp.Precision[] precisions = Timestamp.Precision.values();
+            if (timestampTemplate != "random"){
+                while (size > 0) {
+                    IonReaderBuilder readerBuilder = IonReaderBuilder.standard();
+                    IonReader reader = readerBuilder.build(timestampTemplate);
+                    reader.next();
+                    reader.stepIn();
+                    reader.next();
```

Comment:
Rather than re-reading the timestamp templates on every iteration of the `while` loop (which requires a lot of unnecessary duplicated effort), consider reading the templates once before entering the loop and accessing them as needed from within the loop.